src/library/scala/collection/immutable/ListSet.scala

```diff
@@ -110,10 +112,13 @@ object ListSet extends IterableFactory[ListSet] {
   def from[E](it: scala.collection.IterableOnce[E]): ListSet[E] =
     it match {
       case ls: ListSet[E] => ls
+      case _ if it.knownSize == 0 => empty[E]
       case _ => (newBuilder[E] ++= it).result()
     }
 
-  private object EmptyListSet extends ListSet[Any]
+  private object EmptyListSet extends ListSet[Any] {
+    override def knownSize: Int = 0
```

Comment:
override isEmpty? And def iterator = Iterator.empty?

Reply:
We actually inherit `override def isEmpty: Boolean = true`. So actually the override of `knownSize` is redundant. `Iterator.empty` sounds like a good idea.

Reply:
@retronym adding an explicit override for isEmpty is not bad since it enforces the semantics of the object.